app/src/main/java/org/oppia/android/app/customview/ContinueButtonView.kt

```diff
@@ -0,0 +1,129 @@
+package org.oppia.android.app.customview
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.animation.AnimationUtils
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.oppia.android.R
+import org.oppia.android.app.model.EphemeralState
+import org.oppia.android.app.player.state.itemviewmodel.ContinueInteractionViewModel
+import org.oppia.android.app.utility.LifecycleSafeTimerFactory
+import org.oppia.android.app.view.ViewComponentFactory
+import org.oppia.android.app.view.ViewComponentImpl
+import org.oppia.android.domain.exploration.ExplorationProgressController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.platformparameter.EnableContinueButtonAnimation
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.system.OppiaClock
+import javax.inject.Inject
+
+class ContinueButtonView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = R.style.StateButtonActive
+) : androidx.appcompat.widget.AppCompatButton(context, attrs, defStyleAttr) {
+
+  @field:[Inject EnableContinueButtonAnimation]
+  lateinit var enableContinueButtonAnimation: PlatformParameterValue<Boolean>
+
+  @Inject
+  lateinit var fragment: Fragment
+
+  @Inject
+  lateinit var oppiaClock: OppiaClock
+
+  @Inject
+  lateinit var explorationProgressController: ExplorationProgressController
+
+  @Inject
+  lateinit var lifecycleSafeTimerFactory: LifecycleSafeTimerFactory
+
+  private lateinit var viewModel: ContinueInteractionViewModel
+  private var isAnimationTimerFinished = false
+
+  val animateContinueButton = MutableLiveData(false)
```

Comment:
Can this not be an ObservableBoolean, instead?

Reply:
this needs to be a live data, as its value is observed for triggering an animation